libtiledbsoma/src/soma/soma_array.h

```diff
@@ -1031,6 +1038,55 @@ class SOMAArray : public SOMAObject {
      */
     std::vector<int64_t> maxshape();
 
+    /**
+     * This wires up to Python/R to tell a user if they can call resize() on an
+     * array without error. For single arrays, they could just call resize() and
+     * take their chances -- but for experiment-level resize (e.g. append mode)
+     * it's crucial that we provide a can-we-do-them-all pass through all arrays
+     * in the experiment before attempting any of them.
+     *
+     * On failure, returns false and an error string suitable for showing
+     * to the user; on success, returns true and the empty string.
```

Comment:
Just mulling over this. I feel like returning an error code is better (maybe using enums) rather than returning a pair,  if we don't want to throw exceptions.

Reply:
These are error strings directly to the user without needing extra decode separately in Python/R.